### PR TITLE
LimitReader copies bytes instead of slicing so the underlying big array can be freed

### PR DESCRIPTION
**Pull request by kvch**
In `LimitReader` when truncating the line if the configured limit was reached, the message was sliced. Slicing does not touch the underlying array. To make sure the long line can be truncated and freed, the reader copies the contents to a new array. Thus, freeing the data.

### CODE REVIEW

File: libbeat/reader/readfile/limit.go [hunk 1]
```diff
@@ -37,7 +39,10 @@ func NewLimitReader(r reader.Reader, maxBytes int) *LimitReader {
 func (r *LimitReader) Next() (reader.Message, error) {
 	message, err := r.reader.Next()
 	if len(message.Content) > r.maxBytes {
-		message.Content = message.Content[:r.maxBytes]
+		n = copy(message.Content, message.Content[:r.maxBytes])
```
Comment (urso) [inline libbeat/reader/readfile/limit.go]: `copy` doesn't allocate or change the size of the underlying array.

it must say:

```
tmp := make([]byte, len(message.Content))
copy(tmp, message.Content)
message.Content = tmp
```

**Comment by kvch** [comment]
Fixed and added changelog.

File: CHANGELOG.next.asciidoc [hunk 2]
```diff
@@ -73,6 +73,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 - Fix goroutine leak happening when harvesters are dynamically stopped. {pull}11263[11263]
 - Fix `add_docker_metadata` source matching, using `log.file.path` field now. {pull}11577[11577]
 - Add missing Kubernetes metadata fields to Filebeat CoreDNS module, and fix a documentation error. {pull}11591[11591]
+- LimitReader copies bytes instead of slicing so the underlying big array can be freed. {pull}11524[11524]
```
Comment (urso) [inline CHANGELOG.next.asciidoc]: In user facing changelog better mention what has been fixed, not how an implementation detailt has been changed. This helps users to identify if they've been affected.

How about:

`Reduce memory usage if long lines are truncated to fit `max_bytes` limit. The line buffer is copied into a smaller buffer now. This allows the runtime to release unused memory more early.`.

**Comment by kvch** [comment]
Only reworded changelog entry, merging as before everything was green.

**Pull request merged**